tests/integration/cmor/_fixes/cmip6/test_gfdl_esm4.py

```diff
@@ -0,0 +1,79 @@
+"""Tests for the fixes of GFDL-ESM4."""
+import os
+
+import iris
+import numpy as np
+import pytest
+from cf_units import Unit
+from netCDF4 import Dataset
+
+from esmvalcore.cmor._fixes.cmip6.gfdl_esm4 import Siconc
+from esmvalcore.cmor.fix import Fix
+from esmvalcore.cmor.table import get_var_info
+
+
+@pytest.fixture
+def siconc_file(tmp_path):
+    """Create netcdf file with similar issues as ``cl``."""
+    nc_path = os.path.join(tmp_path, 'gfdl_esm4_siconc.nc')
+    dataset = Dataset(nc_path, mode='w')
+    dataset.createDimension('time', size=1)
+    dataset.createDimension('lat', size=1)
+    dataset.createDimension('lon', size=1)
+
+    # Dimensional variables
+    dataset.createVariable('time', np.float64, dimensions=('time',))
+    dataset.createVariable('lat', np.float64, dimensions=('lat',))
+    dataset.createVariable('lon', np.float64, dimensions=('lon',))
+    dataset.variables['time'][:] = [0.0]
+    dataset.variables['time'].standard_name = 'time'
+    dataset.variables['time'].units = 'days since 6543-2-1'
+    dataset.variables['lat'][:] = [-30.0]
+    dataset.variables['lat'].standard_name = 'latitude'
+    dataset.variables['lat'].units = 'degrees_north'
+    dataset.variables['lon'][:] = [30.0]
+    dataset.variables['lon'].standard_name = 'longitude'
+    dataset.variables['lon'].units = 'degrees_east'
+    dataset.createVariable('siconc', np.float64,
+                           dimensions=('time', 'lat', 'lon'))
+    dataset.variables['siconc'][:] = 22.
+    dataset.variables['siconc'].standard_name = 'sea_ice_area_fraction'
+    dataset.variables['siconc'].units = '%'
+
+    dataset.close()
```

Comment:
```suggestion
with Dataset(nc_path, mode='w') as dataset:
    dataset.createDimension('time', size=1)
    dataset.createDimension('lat', size=1)
    dataset.createDimension('lon', size=1)

    # Dimensional variables
    dataset.createVariable('time', np.float64, dimensions=('time',))
    dataset.createVariable('lat', np.float64, dimensions=('lat',))
    dataset.createVariable('lon', np.float64, dimensions=('lon',))
    dataset.variables['time'][:] = [0.0]
    dataset.variables['time'].standard_name = 'time'
    dataset.variables['time'].units = 'days since 6543-2-1'
    dataset.variables['lat'][:] = [-30.0]
    dataset.variables['lat'].standard_name = 'latitude'
    dataset.variables['lat'].units = 'degrees_north'
    dataset.variables['lon'][:] = [30.0]
    dataset.variables['lon'].standard_name = 'longitude'
    dataset.variables['lon'].units = 'degrees_east'
    dataset.createVariable('siconc', np.float64,
                           dimensions=('time', 'lat', 'lon'))
    dataset.variables['siconc'][:] = 22.
    dataset.variables['siconc'].standard_name = 'sea_ice_area_fraction'
    dataset.variables['siconc'].units = '%'
```
using the context manager is recommended, this ensures that `dataset.close()` is called even if an exception is raised before reaching the `dataset.close()` statement.